lib/logger.js

```diff
@@ -8,15 +8,12 @@ function Logger() {
 	Object.keys(Logger.prototype).forEach(function (key) {
 		this[key] = this[key].bind(this);
 	}, this);
+
+	this.reporter = reporter;
```

Comment:
How about this, in `cli.js`:

```
if (cli.flags.api) {
	reporter = tapReporter(api);
}
[...]
reporter.api = api;
logger = new Logger(reporter);
```

I think that reads more clearly, and the `Logger` is no longer manipulating the reporter behind the scenes.